src/hpc/threadapi.c

```diff
@@ -2011,8 +2015,14 @@ MigrateObjects(int count, Obj * objects, Region * target, int retype)
                 return 0;
         }
     }
-    for (i = 0; i < count; i++)
-        SET_REGION(objects[i], target);
+    for (i = 0; i < count; i++) {
+        Obj obj = objects[i];
+        switch (TNUM_OBJ(obj)) {
+        case T_PREC:
+          SortPRecRNam(obj, 0);
```

Comment:
Is this meant to be "future proof", or why isn't this just a simple `if (TNUM_OBJ(obj) == T_PREC)` check?

If it stays as a switch/case, I'd insert a `break`; here

Reply:
This is a relic of an earlier version, I think, and should be fixed.